### PR TITLE
feature: add Relative Strength Index (RSI) indicator

### PR DESCRIPTION
**Pull request by narumiruna**

### CODE REVIEW

File: pkg/indicator/rsi.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+	var avgLoss float64
+	if len(inc.Prices) == inc.Window+1 {
+		diffValues := inc.Prices.Diff()
+
+		avgGain = diffValues.PositiveValues().AbsoluteValues().Sum() / float64(inc.Window)
+		avgLoss = diffValues.NegativeValues().AbsoluteValues().Sum() / float64(inc.Window)
+	} else {
+		diff := price - inc.Prices[len(inc.Prices)-2]
+		currentGain := math.Abs(math.Max(diff, 0))
+		currentLoss := math.Abs(math.Min(diff, 0))
```
Comment (zenixls2) [inline pkg/indicator/rsi.go:45]: math.Abs could be replaced by -.

File: pkg/indicator/rsi.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+	var avgGain float64
+	var avgLoss float64
+	if len(inc.Prices) == inc.Window+1 {
+		diffValues := inc.Prices.Diff()
+
+		avgGain = diffValues.PositiveValues().AbsoluteValues().Sum() / float64(inc.Window)
+		avgLoss = diffValues.NegativeValues().AbsoluteValues().Sum() / float64(inc.Window)
+	} else {
+		diff := price - inc.Prices[len(inc.Prices)-2]
+		currentGain := math.Abs(math.Max(diff, 0))
```
[the remark this answers is not in the document]
Comment (narumiruna) [reply pkg/indicator/rsi.go:44]: ok

File: pkg/types/float_slice.go [hunk 3]
```diff
@@ -53,3 +53,55 @@ func (s Float64Slice) Tail(size int) Float64Slice {
...
+		if i == 0 {
+			values.Push(0)
+			continue
+		}
+		values.Push(v - s[i-1])
+	}
+	return values
+}
+
+func (s Float64Slice) PositiveValues() Float64Slice {
```
Comment (zenixls2) [inline pkg/types/float_slice.go:69]: PositiveValueOrZero

Comment (narumiruna) [reply pkg/types/float_slice.go:69]: that's better

File: pkg/types/float_slice.go [hunk 4]
```diff
@@ -53,3 +53,55 @@ func (s Float64Slice) Tail(size int) Float64Slice {
...
+	var values Float64Slice
+	for _, v := range s {
+		values.Push(math.Max(v, 0))
+	}
+	return values
+}
+
+func (s Float64Slice) NegativeValues() Float64Slice {
```
Comment (zenixls2) [inline pkg/types/float_slice.go:77]: NegativeValuesOrZero

**Review by c9s — APPROVED** [review]

**Pull request merged**